File: fast-tax-nodejs-examples/get_best_match_rest_sdk_example.js
import { GetBestMatchClient } from '../fast-tax-nodejs/REST/get_best_match_rest.js';

async function GetBestMatchRestGo(licenseKey, isLive) {
    console.log("\n---------------------------------");
    console.log("FastTax - GetBestMatch - REST SDK");
    console.log("---------------------------------");

    const address = "27 E Cota St";
    const address2 = "";
    const city = "Santa Barbara";
    const state = "CA";
    const zip = "93101";
    const taxType = "sales";
    const timeoutSeconds = 15;

    console.log("\n* Input *\n");
    console.log(`Address        : ${address}`);
    console.log(`Address2       : ${address2}`);
    console.log(`City           : ${city}`);
    console.log(`State          : ${state}`);
    console.log(`Zip            : ${zip}`);
    console.log(`Tax Type       : ${taxType}`);
    console.log(`License Key    : ${licenseKey}`);
    console.log(`Is Live        : ${isLive}`);
    console.log(`Timeout Seconds: ${timeoutSeconds}`);

    try {
        const response = await GetBestMatchClient.invoke(
            address,
            address2,
            city,
            state,
            zip,
            taxType,
            licenseKey,
            isLive,
            timeoutSeconds
        );

        console.log("\n* Tax Info *\n");
        if (response) {
            console.log(`Match Level  : ${response.MatchLevel}`);
            if (response.TaxInfoItems && response.TaxInfoItems.length > 0) {
                for (const taxInfo of response.TaxInfoItems) {
                    console.log(`Zip              : ${taxInfo.Zip}`);
                    console.log(`City             : ${taxInfo.City}`);
                    console.log(`County           : ${taxInfo.County}`);
                    console.log(`State Abbrev     : ${taxInfo.StateAbbreviation}`);
                    console.log(`State Name       : ${taxInfo.StateName}`);
                    console.log(`Total Tax Rate   : ${taxInfo.TaxRate}`);
                    console.log(`State Rate       : ${taxInfo.StateRate}`);
                    console.log(`City Rate        : ${taxInfo.CityRate}`);
                    console.log(`County Rate      : ${taxInfo.CountyRate}`);
                    console.log(`County Dist Rate : ${taxInfo.CountyDistrictRate}`);
                    console.log(`City Dist Rate   : ${taxInfo.CityDistrictRate}`);
                    console.log(`Special Dist Rate: ${taxInfo.SpecialDistrictRate}`);
                    console.log(`Total Tax Exempt : ${taxInfo.TotalTaxExempt}`);
                    console.log(`Notes Codes      : ${taxInfo.NotesCodes}`);
                    console.log(`Notes Desc       : ${taxInfo.NotesDesc}`);

                    console.log("\n* Information Components *\n");
                    if (taxInfo.InformationComponents && taxInfo.InformationComponents.length > 0) {
                        taxInfo.InformationComponents.forEach((component) => {
                            console.log(`${component.Name}: ${component.Value}`);
                        });
                    } else {
                        console.log("No information components found.");
                    }
                }
            } else {
                console.log("No tax info items found.");
            }
        } else {
            console.log("No tax info found.");
        }

        if (response.Error) {
            console.log("\n* Error *\n");
            console.log(`Error Desc    : ${response.Error.Desc}`);
            console.log(`Error Number  : ${response.Error.Number}`);
            console.log(`Error Location: ${response.Error.Location}`);
            return;
        }
    } catch (e) {
        console.log("\n* Error *\n");
        console.log(`Error Message: ${e.message}`);
    }
}

export { GetBestMatchRestGo };

File: fast-tax-python-examples/get_best_match_rest_sdk_example.py
import sys
import os

sys.path.insert(0, os.path.abspath("../fast-tax-python/REST"))

from get_best_match_rest import get_best_match


def get_best_match_rest_sdk_go(is_live: bool, license_key: str) -> None:
    print("\r\n---------------------------------")
    print("FastTax - GetBestMatch - REST SDK")
    print("---------------------------------")

    address = "27 E Cota St"
    address2 = ""
    city = "Santa Barbara"
    state = "CA"
    zip = "93101"
    tax_type = "sales"

    print("\r\n* Input *\r\n")
    print(f"Address    : {address}")
    print(f"Address2   : {address2}")
    print(f"City       : {city}")
    print(f"State      : {state}")
    print(f"Zip        : {zip}")
    print(f"Tax Type   : {tax_type}")
    print(f"License Key: {license_key}")
    print(f"Is Live    : {is_live}")

    try:
        response = get_best_match(address, address2, city, state, zip, tax_type, license_key, is_live)

        print("\r\n* Tax Info *\r\n")
        if response and not response.Error:
            print(f"Match Level      : {response.MatchLevel}")
            if response.TaxInfoItems and len(response.TaxInfoItems) > 0:
                for tax_info in response.TaxInfoItems:
                    print(f"Zip              : {tax_info.Zip}")
                    print(f"City             : {tax_info.City}")
                    print(f"County           : {tax_info.County}")
                    print(f"State Abbrev     : {tax_info.StateAbbreviation}")
                    print(f"State Name       : {tax_info.StateName}")
                    print(f"Total Tax Rate   : {tax_info.TaxRate}")
                    print(f"State Rate       : {tax_info.StateRate}")
                    print(f"City Rate        : {tax_info.CityRate}")
                    print(f"County Rate      : {tax_info.CountyRate}")
                    print(f"County Dist Rate : {tax_info.CountyDistrictRate}")
                    print(f"City Dist Rate   : {tax_info.CityDistrictRate}")
                    print(f"Special Dist Rate: {tax_info.SpecialDistrictRate}")
                    print(f"Total Tax Exempt : {tax_info.TotalTaxExempt}")
                    print(f"Notes Codes      : {tax_info.NotesCodes}")
                    print(f"Notes Desc       : {tax_info.NotesDesc}")

                    print("\r\n* Information Components *\r\n")
                    if tax_info.InformationComponents and len(tax_info.InformationComponents) > 0:
                        for component in tax_info.InformationComponents:
                            print(f"{component.Name}: {component.Value}")
                    else:
                        print("No information components found.")
            else:
                print("No tax info items found.")
        else:
            print("No tax info found.")

        if response.Error:
            print("\r\n* Error *\r\n")
            print(f"Error Desc    : {response.Error.Desc}")
            print(f"Error Number  : {response.Error.Number}")
            print(f"Error Location: {response.Error.Location}")

    except Exception as e:
        print("\r\n* Error *\r\n")
        print(f"Error Message: {str(e)}")
